### PR TITLE
Remove panics in tz datagen

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

File: provider/source/src/time_zones/convert.rs [hunk 1]
```diff
@@ -407,67 +409,67 @@ macro_rules! long_short_impls {
...
-                                                        }
-                                                    })
-                                                    .collect::<Vec<_>>(),
-                                            }
-                                        })
+                                .filter_map(|(key, metazone)| {
+                                    metazone.$field.as_ref().and_then(type_fallback).and_then(
+                                        |format| {
+                                            Some((
+                                                meta_zone_id_data.get(key)?.clone(),
```
Comment (sffc) [inline provider/source/src/time_zones/convert.rs:422]: Question: this one now drops metazones instead of panicking? And similar elsewhere?

**Review by sffc — APPROVED** [review]
It seems fine, but I have questions:

1. What problem is it fixing? We're generally strict in datagen, and this appears to make us less strict. (Please understand my perspective: you send me a 200-line PR with no description and no context, and I am supposed to make sense of it.)
2. Why is there a diff in the baked data but not the JSON data? Presumably there's some locale somewhere that is impacted by this?

**Comment by sffc** [comment]
(I approve because I trust your judgement here, but I would like answers to my questions)

File: provider/source/src/time_zones/convert.rs (hunk 1 above)
Comment (robertbastian) [reply provider/source/src/time_zones/convert.rs:422]: Yes

**Comment by robertbastian** [comment]
This is meant to set the stage for one or two PRs that don't work with the panicking implementation.

This seems to impact some basic locales.

**Pull request merged**